### ISSUE TITLE
Cannot build a project with #include "ArduinoJson.h" in Atmel Studio 7

### ISSUE THREAD

**Issue by martingcavallo** [bblanchon/ArduinoJson#1112]
Hi,

I am not able to build a project in Atmel Studio 7 which include arduinoJson library. When I buid it, I get errors in relation with the #include directives that are inside the own library. Whta's more, I created a new project in which I only included the arduinoJson library and I left the main function empty and I got the same errors. I tried to add the library in the different ways, but the result is same.

I mean, the compiler does not find the files. The error descriptions say "xxxxxxx.h:No such file or directory." 

In some case, it seems like the solution in add .h extension into the name of the file, but it is not the solution in other cases.

Have anybody got an idea? I attach some pictures for you.

Thanks in advance.

![ostream](https://user-images.githubusercontent.com/56684118/67011539-3717ac80-f0c6-11e9-9a71-cfd71a992c29.PNG)
![ostream2](https://user-images.githubusercontent.com/56684118/67011540-3717ac80-f0c6-11e9-9685-c185d7522bfa.PNG)
![String](https://user-images.githubusercontent.com/56684118/67011542-3717ac80-f0c6-11e9-802c-1f8a95ec72e8.PNG)
![String2](https://user-images.githubusercontent.com/56684118/67011543-37b04300-f0c6-11e9-8604-d2c08a600d87.PNG)
![String3](https://user-images.githubusercontent.com/56684118/67011544-37b04300-f0c6-11e9-814f-66ba7e8de500.PNG)

**Comment by bblanchon** [comment]
Hi @mgcavallo,

ArduinoJson looks for `<string>` and `<ostream>` because it incorrectly guessed that the target is a computer (as opposed to a microcontroller).

Try to set `ARDUINOJSON_EMBEDDED_MODE` to 1:

```c++
#define ARDUINOJSON_EMBEDDED_MODE 1
#include "ArduinoJson-v6.12.0.h"
```

This will disable support for `std::string` and `std::ostream`, which should fix your problem.

There is probably a way to correctly detect the target type, I'll have a look ASAP.

Best Regards,
Benoit

**Comment by martingcavallo** [comment]
Hi @bblanchon ,

Thank you for your prompt response.
That's right! The #define directive setting ARDUINOJSON_EMBEDDED_MODE to 1, resolve my issue.

However, I agree with you that there would be a way to detect the target type.

Best Regards,
Martin

**Comment by martingcavallo** [comment]
Hi, 

I am having a new issue. I think it is a silly issue, but I do not unerstand.
I am migrating from ArduinoJson5 to ArduinoJson6 and I have the problem with serialization. The error said that no match function for the arguments. I want to serialize a StaticJsonDocument into a char array. The documentation shows that it is possible.

Attach pictures.

![Migracion a JS6](https://user-images.githubusercontent.com/56684118/67099974-4285d900-f195-11e9-8cb1-e3cecaaeb8de.PNG)
![serializeJson](https://user-images.githubusercontent.com/56684118/67099975-4285d900-f195-11e9-8e96-03559d1231a3.PNG)

**Comment by martingcavallo** [comment]
Hi,

I could solve the problem adding the lenght as the third parameter.

Regards,

Martin

**Comment by bblanchon** [comment]
Fix published in [ArduinoJson 6.13.0](https://github.com/bblanchon/ArduinoJson/releases/tag/v6.13.0)

**Comment by martingcavallo** [comment]
Thank you Benoit!!! I will try verison 6.13.0 ASAP.
